src/backend/distributed/commands/dependencies.c

```diff
@@ -0,0 +1,225 @@
+	 * the pg_dist_object record becomes visible.
+	 */
+	LockRelationOid(DistNodeRelationId(), RowShareLock);
+	workerNodeList = ActivePrimaryNodeList();
+	if (list_length(workerNodeList) <= 0)
+	{
+		/* no nodes to execute on */
+		return;
+	}
+	foreach(workerNodeCell, workerNodeList)
+	{
+		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+		MultiConnection *connection = NULL;
+
+		char *nodeName = workerNode->workerName;
+		uint32 nodePort = workerNode->workerPort;
+
+		connection = StartNodeUserDatabaseConnection(connectionFlag, nodeName, nodePort,
+													 CitusExtensionOwnerName(), NULL);
+
+		connections = lappend(connections, connection);
+	}
+	FinishConnectionListEstablishment(connections);
+
+	/*
+	 * create dependency on all nodes
+	 */
+	foreach(connectionCell, connections)
+	{
+		MultiConnection *connection = (MultiConnection *) lfirst(connectionCell);
+		ExecuteCriticalRemoteCommandList(connection, ddlCommands);
+	}
+
+	/*
+	 * mark all objects as distributed
+	 */
+	foreach(dependencyCell, dependencies)
+	{
+		ObjectAddress *dependency = (ObjectAddress *) lfirst(dependencyCell);
+		MarkObjectDistributed(dependency);
+	}
+
+	/*
+	 * disconnect from nodes
+	 */
+	foreach(connectionCell, connections)
+	{
+		MultiConnection *connection = (MultiConnection *) lfirst(connectionCell);
+		CloseConnection(connection);
+	}
+}
+
+
+/*
+ * GetDependencyCreateDDLCommands returns a list (potentially empty or NIL) of ddl
+ * commands to execute on a worker to create the object.
+ */
+static List *
+GetDependencyCreateDDLCommands(const ObjectAddress *dependency)
+{
+	switch (getObjectClass(dependency))
+	{
+		case OCLASS_SCHEMA:
+		{
+			const char *schemaDDLCommand = CreateSchemaDDLCommand(dependency->objectId);
+
+			if (schemaDDLCommand == NULL)
+			{
+				/* no schema to create */
+				return NIL;
+			}
+
+			return list_make1((void *) schemaDDLCommand);
+		}
+
+		default:
+		{
+			break;
+		}
+	}
+
+	/*
+	 * make sure it fails hard when in debug mode, leave a hint for the user if this ever
+	 * happens in production
+	 */
+	Assert(false);
+	ereport(ERROR, (errmsg("unsupported object %s for distribution by citus",
+						   getObjectTypeDescription(dependency)),
+					errdetail(
+						"citus tries to recreate an unsupported object on its workers"),
+					errhint("please report a bug as this should not be happening")));
+	return NIL;
+}
+
+
+/*
+ * ReplicateAllDependenciesToNode replicate all previously marked objects to a worker node
+ */
+void
+ReplicateAllDependenciesToNode(const char *nodeName, int nodePort)
+{
+	const uint64 connectionFlag = FORCE_NEW_CONNECTION;
+	ListCell *dependencyCell = NULL;
+	List *dependencies = NIL;
+	List *ddlCommands = NIL;
+	MultiConnection *connection = NULL;
+
+	/*
+	 * collect all dependencies in creation order and get their ddl commands
+	 */
+	dependencies = GetDistributedObjectAddressList();
+	if (list_length(dependencies) > 100)
```

Comment:
Super minor: Can you maybe mention that 100 is an arbitrary value, where we start to see that the dependency replications might take some that that is not negligible